stack.go

```diff
@@ -6,47 +6,80 @@ import (
 	"strings"
 )
 
+// Stack is an array of stack frames stored in a human readable format.
+type Stack []StackFrame
+
+// contains checks if a stack trace contains a specific frame.
+func (s *Stack) contains(frame StackFrame) bool {
+	for _, f := range *s {
+		if f == frame {
+			return true
+		}
+	}
+	return false
+}
+
+// insertFrames inserts a wrap error frame into the correct place of the root error stack trace.
+func (s *Stack) insertFrame(wFrames []StackFrame) {
+	if len(wFrames) == 0 || s.contains(wFrames[0]) {
+		return
+	} else if len(wFrames) == 1 {
+		// append the frame to the end if there's only one
+		*s = append(*s, wFrames[0])
+	}
+
+	rStack := *s
+	for i, f := range rStack {
```

Comment:
if we are looping over stack here then that s.contains can be avoided in ln24 and can be done in this loop by checking if f == wFrame[0] then return

Reply:
fixed it, that actually allowed me to delete the contains method